river/drift/d3.py

```diff
@@ -0,0 +1,189 @@
+import numpy as np
+from sklearn.linear_model import LogisticRegression
+
+from river import metrics
+from river.base import DriftDetector
+
+
+class D3(DriftDetector):
+    r"""Drift Detection Method
+
+    D3 (Discriminative Drift Detector) is an unsupervised drift detection
+    method which uses a discriminative classifier that can be used with any
+    online algorithm without a built-in drift detector. It holds a fixed
+    size sliding window of the latest data having two sets: the old and the
+    new. A simple classifier is trained to distinguish these sets. It
+    detects a drift with respect to classifier performance (AUC).
+
```

Comment:
Consider mentioning that this method looks for drift in multi-dimensional data. To differentiate it from the rest of the methods that work on uni-dimensional data.

Reply:
Thanks for your comments and suggestions. I'll update the PR till the end of this month. I'm a bit busy right now due to work.

Reply:
That is alright. Thanks for looking into this.